sklearn/utils/enet_proj.py

```diff
@@ -0,0 +1,59 @@
+__author__ = 'arthur'
+
+import numpy as np
+from numpy import sqrt
+from sklearn.utils import check_random_state
+
+def _enet_norm_for_proj(b, gamma):
+    return np.sum(b * (1 + gamma / 2 * b))
+
+
+def enet_norm(b, l1_ratio=0.1):
+    if l1_ratio == 0:
+        return sqrt(np.sum(b ** 2))
+    b_abs = np.abs(b)
+    return np.sum(b_abs * (l1_ratio + (1 - l1_ratio) * b_abs))
+
+
+def enet_proj(b, tau=1, l1_ratio=0.1, random_state=None):
```

Comment:
This code is both interesting an non trivial. Please add a reference to the paper that best describe it. Also please add unit tests dedicated to check the correctness of this implementation on a bunch of (deterministically seeded) random vectors with a elastic net radius scaled to be large enough to lie outside of the ball defined by `l1_ratio` and `tau`. 

Actually one should find a more explicit name for the `tau` variable. For instance `radius`.

Also please use explicit function names such as `enet_projection` and finally make those functions private (`_enet_projection`).